core/client/src/main/java/alluxio/client/UfsBlockReader.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client;
+
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.netty.NettyUfsBlockReader;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+
+/**
+ * The interface to read an under file system file through a worker's data server.
+ */
+public interface UfsBlockReader extends Closeable {
+
+  /**
+   * The factory for the {@link UfsBlockReader}.
+   */
+  class Factory {
+
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory for {@link UfsBlockReader}.
+     *
+     * @param context the file system context
+     * @return a new instance of {@link UfsBlockReader}
+     */
+    public static UfsBlockReader create(FileSystemContext context) {
+      return new NettyUfsBlockReader(context);
+    }
+  }
+
+  /**
+   * Reads a UFS block with a offset and length.
```

Comment:
not sure "ufs block" is meaningful here, as ufs may not have the concept of block. Maybe "Reads a block from UFS with the offset and length"?